BEIMA.Client/src/pages/Maps/MapPage.js

```diff
@@ -0,0 +1,67 @@
+import React, {useState, useEffect} from 'react';
+import styles from './MapPage.module.css'
+import GetDeviceList from '../../services/GetDeviceList';
+import { useOutletContext, useNavigate } from 'react-router-dom';
+import ReactMapGL, {Marker} from 'react-map-gl';
+import 'mapbox-gl/dist/mapbox-gl.css';
+
+/**
+ * React component for creating the map page.
+ * @returns The map page content.
+ */
+const MapPage = () => {
+  const [devices, setDevices] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [setPageName] = useOutletContext();
+
+  const [viewport, setViewport] = useState({
+    latitude: 43.60361004262107,
+    longitude: -116.20004380745182,
+    zoom: 15
+  });
+
+  const navigate = useNavigate();
+
+  const DeviceListCall = async () => {
+    let data = await GetDeviceList();
+    return data.response;
+  }
+
+  useEffect(() => {
+    setPageName('Map')
```

Comment:
We can add this to the useEffect below as the first line, make sure to also add it to the dependency array